src/tasks/LibraryBuilder/Templates/CMakeLists.txt.template

```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.10)
+
+project(%LIBRARY_NAME%)
+
+enable_language(%CMAKE_LANGS%)
+
+set(DOTNET_AOT_SOURCES
+    %AotSources%
+)
+set(DOTNET_AOT_OBJECTS
+    %AotObjects%
+)
+set(DOTNET_EXTRA_SOURCES
+    %ExtraSources%
+    autoinit.c
+)
+
+include_directories("%MonoInclude%")
+
+add_library(
+    aot_library STATIC
+    ${DOTNET_AOT_SOURCES}   
+)
+target_link_libraries(
+    aot_library
+    ${DOTNET_AOT_OBJECTS}
```

Comment:
There is a push in https://github.com/dotnet/runtime/pull/81924 to have explicit scope keywords `PUBLIC|PRIVATE|INTERFACE` supplied for each call to `target_link_libraries`, we should add them here as well to conform.